source/contracts/trading/ClaimProceeds.sol

```diff
@@ -28,7 +28,11 @@ contract ClaimProceeds is CashAutoConverter, ReentrancyGuard, IClaimProceeds {
         for (uint8 _outcome = 0; _outcome < _market.getNumberOfOutcomes(); ++_outcome) {
             IShareToken _shareToken = _market.getShareToken(_outcome);
             uint256 _numberOfShares = _shareToken.balanceOf(msg.sender);
-            var (_shareHolderShare, _creatorShare, _reporterShare) = divideUpWinnings(_market, _winningReportingToken, _outcome, _numberOfShares);
+            var (_proceeds, _shareHolderShare, _creatorShare, _reporterShare) = divideUpWinnings(_market, _winningReportingToken, _outcome, _numberOfShares);
+
+            if (_proceeds > 0) {
+                _market.getUniverse().decrementOpenInterest(_proceeds);
+            }
```

Comment:
🤔 It would be more correct (game theory) to decrease open interest for all outstanding shares in a market when it is finalized.  This is likely not feasible, and we have to be _really_ careful to then make it so closing a complete set after a market is finalized doesn't decrease open interest (as that would double-count open interest decreases).

In general, the open interest calculations are _super_ scary to me since they are critical to the security model and they don't reset each reporting period/market like a lot of values.  This means that a subtle bug that slowly bleeds off open interest or slowly accumulates open interest could result in a long-term problem.  This may even be a problem with rounding errors (superman 2 attack).

So, unless you think the above thought is easy to implement and provide strong guarantees on, I'm OK with this resulting in a bit of an over-estimate of open interest.  It does introduce a different problem where over time, open interest will constantly grow (indefinitely) as money gets forgotten/lost in markets.

Reply:
I think it's probably doable. I'll attempt it and if it feels fragile we can go back to this

Reply:
Be incredibly careful with it.  If your gut tells you that it is at all risky I would say fall back to this.  Tracking open interest accurately over an indefinite period of time is a big fear of mine.

Reply:
I have a pattern in mind for doing this but implementing even a portion of it puts the `Market` contract over the size limit :/ I added a CONSIDER comment so maybe if/when we address Market in some larger capacity we can add this.